File: two-sum 07-04-2024.js
var twoSum = function(nums, target) {
  const pairIdx = {};

  for (let i = 0; i < nums.length; i++) {
      const num = nums[i];
      if (target - num in pairIdx) {
          return [i, pairIdx[target - num]];
      }
      pairIdx[num] = i;
  }
};
/*Only thing I wonder with this code is what if it uses itself twice? */

File: two-sum 07-04-2024.py
def twoSum(nums, target):
  numMap = {}
  n = len(nums)
  for i in range(n):
      numMap[nums[i]] = i
  for i in range(n):
     complement = target - nums[i]
     if complement in numMap and numMap[complement] != i:
        return [i, numMap[complement]]
  return []

  '''
  So apparently a hash table is the same as a dictionary in python which makes sense
  this would have a time complexity of n since it only loops through one for loop.
  '''